src/libraries/Microsoft.PowerFx.Core/Public/Values/CompileTimeTypeWrapperRecordValue.cs

```diff
@@ -21,6 +21,12 @@ public static RecordValue AdjustType(RecordType expectedType, RecordValue inner)
                 return inner;
             }
 
+            // If the inner value is an InMemoryRecordValue, we can just adjust the type without recreating field dictionary.
+            if (inner is InMemoryRecordValue inmemoryRecordValue)
```

Comment:
> if (inner is InMemoryRecordValue inmemoryRecordValue) [](http://example.com/codeflow?start=11&length=54)

Could this lead to an unsafe mutation, especially if inner was shared?

Something like:
With( { t:  .. table ...},  F(t), G(t) )

Reply:
You are correct, I'll change the design.